[10.0-preview.7] Cannot compile using logging source generator due to CS0234 errors

### Description

After updating OpenTelemetry to .NET 10 preview 7, some of our projects fail to compile for due to the following error:

```text
D:\Coding\open-telemetry\opentelemetry-dotnet\test\Benchmarks\obj\Debug\net462\Microsoft.Extensions.Logging.Generators\Microsoft.Extensions.Logging.Generators.LoggerMessageGenerator\LoggerMessage.g.cs(37,24): error CS0234: The type or namespace name 'FormattableString' does not exist in the namespace 'System.Diagnostics.CodeAnalysis' (are you missing an assembly reference?)
```

This appears to be due to this line of code that was added by #117009:

https://github.com/dotnet/runtime/blob/7bc40a5a0a29411981e64733b803f7a16fedbd86/src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Emitter.cs#L180

I think it should be this?

```diff
- "global::System.Diagnostics.CodeAnalysis.FormattableString.Invariant("
+ "global::System.FormattableString.Invariant("
```


### Reproduction Steps

1. Clone https://github.com/open-telemetry/opentelemetry-dotnet/pull/6307/commits/7487f449b98aa49b13e19cfd61c0c602393e757d
2. Run `dotnet build` from the root of the repository


### Expected behavior

The solution compiles.

### Actual behavior

Projects fail to compile due to [`CS0234` errors](https://github.com/open-telemetry/opentelemetry-dotnet/actions/runs/16943136147?pr=6307).

### Regression?

Yes.


### Known Workarounds

1. Don't use the logging source generator
2. Use fewer parameters to not enter the broken code block.


### Configuration

.NET SDK `10.0.100-preview.7.25380.108`

### Other information

_No response_

--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -129,7 +129,6 @@
     <JsonSchemaNetVersion>7.0.2</JsonSchemaNetVersion>
     <NewtonsoftJsonVersion>13.0.3</NewtonsoftJsonVersion>
     <NewtonsoftJsonBsonVersion>1.0.2</NewtonsoftJsonBsonVersion>
-    <SQLitePCLRawbundle_greenVersion>2.0.4</SQLitePCLRawbundle_greenVersion>
     <MoqVersion>4.18.4</MoqVersion>
     <AwesomeAssertionsVersion>8.0.2</AwesomeAssertionsVersion>
     <FsCheckVersion>2.14.3</FsCheckVersion>

--- a/src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Emitter.cs
+++ b/src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Emitter.cs
@@ -177,7 +177,7 @@ namespace {lc.Namespace}
                 string formatMethodBegin =
                     !lm.Message.Contains('{') ? "" :
                     _hasStringCreate ? "string.Create(global::System.Globalization.CultureInfo.InvariantCulture, " :
-                    "global::System.Diagnostics.CodeAnalysis.FormattableString.Invariant(";
+                    "global::System.FormattableString.Invariant(";
                 string formatMethodEnd = formatMethodBegin.Length > 0 ? ")" : "";
 
                 _builder.Append($@"

--- a/src/libraries/Microsoft.Extensions.Logging.Abstractions/tests/Microsoft.Extensions.Logging.Generators.Tests/LoggerMessageGeneratorEmitterTests.cs
+++ b/src/libraries/Microsoft.Extensions.Logging.Abstractions/tests/Microsoft.Extensions.Logging.Generators.Tests/LoggerMessageGeneratorEmitterTests.cs
@@ -3,6 +3,7 @@
 
 using System;
 using System.IO;
+using System.Linq;
 using System.Reflection;
 using System.Threading.Tasks;
 using SourceGenerators.Tests;
@@ -20,7 +21,7 @@ namespace Microsoft.Extensions.Logging.Generators.Tests
             string[] sources = Directory.GetFiles("TestClasses");
             foreach (var src in sources)
             {
-                var testSourceCode = await File.ReadAllTextAsync(src).ConfigureAwait(false);
+                var testSourceCode = File.ReadAllText(src);
 
                 var (d, r) = await RoslynTestUtils.RunGenerator(
                     new LoggerMessageGenerator(),
@@ -253,9 +254,9 @@ namespace Microsoft.Extensions.Logging.Generators.Tests.TestClasses
 
         private async Task VerifyAgainstBaselineUsingFile(string filename, string testSourceCode)
         {
-            string baseline = LineEndingsHelper.Normalize(await File.ReadAllTextAsync(Path.Combine("Baselines", filename)).ConfigureAwait(false));
+            string baseline = LineEndingsHelper.Normalize(File.ReadAllText(Path.Combine("Baselines", filename)));
             string[] expectedLines = baseline.Replace("%VERSION%", typeof(LoggerMessageGenerator).Assembly.GetName().Version?.ToString())
-                                             .Split(Environment.NewLine);
+                                             .Split([Environment.NewLine], StringSplitOptions.None);
 
             var (d, r) = await RoslynTestUtils.RunGenerator(
                 new LoggerMessageGenerator(),
@@ -264,6 +265,13 @@ namespace Microsoft.Extensions.Logging.Generators.Tests.TestClasses
 
             Assert.Empty(d);
             Assert.Single(r);
+
+            if (PlatformDetection.IsNetFramework)
+            {
+                expectedLines = expectedLines.Select(line => line.Replace(
+                    "string.Create(global::System.Globalization.CultureInfo.InvariantCulture, ",
+                    "global::System.FormattableString.Invariant(")).ToArray();
+            }
 
             Assert.True(RoslynTestUtils.CompareLines(expectedLines, r[0].SourceText,
                 out string errorMessage), errorMessage);

--- a/src/libraries/Microsoft.Extensions.Logging.Abstractions/tests/Microsoft.Extensions.Logging.Generators.Tests/Microsoft.Extensions.Logging.Generators.targets
+++ b/src/libraries/Microsoft.Extensions.Logging.Abstractions/tests/Microsoft.Extensions.Logging.Generators.Tests/Microsoft.Extensions.Logging.Generators.targets
@@ -1,7 +1,7 @@
 <Project>
 
   <PropertyGroup>
-    <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
+    <TargetFrameworks>$(NetCoreAppCurrent);$(NetFrameworkCurrent)</TargetFrameworks>
     <EnableDefaultItems>true</EnableDefaultItems>
     <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
     <Nullable>enable</Nullable>
@@ -14,6 +14,10 @@
   <ItemGroup>
     <Compile Include="$(CommonPath)..\tests\SourceGenerators\RoslynTestUtils.cs"
              Link="SourceGenerators\RoslynTestUtils.cs" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
+    <Compile Include="$(CoreLibSharedDir)System\Runtime\CompilerServices\IsExternalInit.cs" />
   </ItemGroup>
 
   <ItemGroup>

--- a/src/libraries/Microsoft.Extensions.Logging.Abstractions/tests/Microsoft.Extensions.Logging.Generators.Tests/Microsoft.Extensions.Logging.Generators.targets
+++ b/src/libraries/Microsoft.Extensions.Logging.Abstractions/tests/Microsoft.Extensions.Logging.Generators.Tests/Microsoft.Extensions.Logging.Generators.targets
@@ -5,7 +5,7 @@
     <EnableDefaultItems>true</EnableDefaultItems>
     <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
     <Nullable>enable</Nullable>
-    <IgnoreForCI Condition="'$(TargetsMobile)' == 'true' or '$(TargetsLinuxBionic)' == 'true' or '$(TargetArchitecture)' == 'ARMv6'">true</IgnoreForCI> 
+    <IgnoreForCI Condition="'$(TargetsMobile)' == 'true' or '$(TargetsLinuxBionic)' == 'true' or '$(TargetArchitecture)' == 'ARMv6'">true</IgnoreForCI>
     <!-- SQLitePCLRaw currently has non-portable RIDs (alpine). Update to newer version once it released.
          https://github.com/ericsink/SQLitePCL.raw/issues/543 -->
     <NoWarn>$(NoWarn);NETSDK1206</NoWarn>
@@ -23,7 +23,6 @@
   <ItemGroup>
     <ProjectReference Include="$(LibrariesProjectRoot)Microsoft.Extensions.Logging.Abstractions\src\Microsoft.Extensions.Logging.Abstractions.csproj" />
     <PackageReference Include="Microsoft.CodeAnalysis" Version="$(RoslynApiVersion)" />
-    <PackageReference Include="SQLitePCLRaw.bundle_green" Version="$(SQLitePCLRawbundle_greenVersion)" />
   </ItemGroup>
 
   <ItemGroup>

